src/ApiCore/Transport/GuzzleTransportTrait.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/*
+ * Copyright 2018, Google Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+namespace Google\ApiCore\Transport;
+
+use Exception;
+use Google\ApiCore\Call;
+use Google\ApiCore\ValidationException;
+use Google\Auth\HttpHandler\HttpHandlerFactory;
+
+/**
+ * An trait for shared functionality between Guzzle HTTP-based transports.
+ */
+trait GuzzleTransportTrait
```

Comment:
Since guzzle is the default transport and users could supply another - WDYT about naming this something like `HTTP11TransportTrait`?

Reply:
Hmm, yep you're right. I'm not that keen on have `11` in the name (purely for aesthetic reasons). Do you think `HttpTransportTrait` is misleading?

Reply:
Agreed there - I have the same reservations. However, `HttpTransportTrait` does seem misleading as it is too broad. I've done a scan of the HTTP/1.1 protocol memo and didn't have any luck finding any other suitable names we could swap in. Pondering alternatives..

Reply:
What about 'HttpUnaryTransportTrait'? That is trying to indicate that this is suitable for transports that support only simple unary rpc methods, not streaming.

Reply:
I like it!

Reply:
Updated, PTAL